File: UI/etheater_mobile/lib/models/show.dart
import 'package:json_annotation/json_annotation.dart';
import 'enums.dart';

part 'show.g.dart';

@JsonSerializable()
class Show {
  int showId;
  String name;
  String description;
  String? picture;
  int duration;
  String? director;
  @JsonKey(fromJson: showGenreFromJson, toJson: showGenreToJson)
  ShowGenre showGenre;
  bool? isDeleted;

  Show(
      {required this.showId,
      required this.name,
      required this.description,
      this.picture,
      required this.duration,
      this.director,
      required this.showGenre,
      this.isDeleted});

  factory Show.fromJson(Map<String, dynamic> json) => _$ShowFromJson(json);
  Map<String, dynamic> toJson() => _$ShowToJson(this);

  static ShowGenre showGenreFromJson(int json) =>
      const ShowGenreConverter().fromJson(json);
  static int showGenreToJson(ShowGenre object) =>
      const ShowGenreConverter().toJson(object);
}

File: eTheater.Model/Show.cs
using eTheater.Model.Enums;
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace eTheater.Model
{
    public class Show
    {
        public int ShowId { get; set; }

        public string Name { get; set; }

        public string Description { get; set; }

        public string? Picture { get; set; }

        public int Duration { get; set; }

        public string? Director { get; set; }

        public ShowGenre? ShowGenre { get; set; }

        public bool IsDeleted { get; set; }
    }
}
